### ISSUE TITLE
Random upload error

### ISSUE THREAD

**Issue by paolocomis** [vimeo/vimeo.py#84]
Hello, I use this library because I had similar problems with the official PHP library (random error: PHP Fatal error: Uncaught exception 'Vimeo \ Exceptions \ VimeoRequestException' with message 'Unable to complete request. [Connect () timed out!] 'In /root/vendor/vimeo/vimeo-api/src/Vimeo/Vimeo.php: 154).

Unfortunately, I'm also having problems with this. Uploading a video often gives me errors:

Traceback (most recent call last):
File "/root/Script/uploadervimeo.py", line 20, in
video_uri = v.upload (video)
File "build/bdist.linux-i686/egg/vimeo/upload.py", line 28, in upload
File "build/bdist.linux-i686/egg/vimeo/upload.py", line 58, in _perform_upload
File "build/bdist.linux-i686/egg/vimeo/upload.py", line 110, in _make_pass
File "build/bdist.linux-i686/egg/vimeo/client.py", line 75, in caller
File "/usr/local/lib/python2.7/site-packages/requests/api.py", line 124, in put
return request('put', url, data=data, *kwargs)
File "/usr/local/lib/python2.7/site-packages/requests/api.py", line 57, in request
return session.request(method=method, url=url, *kwargs)
File "/usr/local/lib/python2.7/site-packages/requests/sessions.py", line 475, in request
resp = self.send(prep, *send_kwargs)
File "/usr/local/lib/python2.7/site-packages/requests/sessions.py", line 585, in send
r = adapter.send(request, *kwargs)
File "/usr/local/lib/python2.7/site-packages/requests/adapters.py", line 453, in send
raise ConnectionError(err, request=request)
requests.exceptions.ConnectionError: ('Connection aborted.', error(104, 'Connection reset by peer'))

Fortunately however sometimes it works regularly but for me it is a big deal.
The script that I created to upload is as follows:

import vimeo

v = vimeo.VimeoClient(
token="XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX",
key="XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX",
secret="XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX")

read1 = open("/root/localvideoname.txt","r")
video = read1.read()
read1.close()

read2 = open("/root/onlinevimeoname.txt","r")
onlinename = read2.read()
read2.close()

video_uri = v.upload (video)

v.patch(video_uri, data={'name': onlinename})

print video_uri

Where am I wrong? your platform is Debian 6. Thanks in advance.

**Comment by paolocomis** [comment]
Hello, I use this library because I had similar problems with the official PHP library (random error: PHP Fatal error: Uncaught exception 'Vimeo \ Exceptions \ VimeoRequestException' with message 'Unable to complete request. [Connect () timed out!] 'In /root/vendor/vimeo/vimeo-api/src/Vimeo/Vimeo.php: 154).

Unfortunately, I'm also having problems with this. Uploading a video often gives me errors:

Traceback (most recent call last):
  File "/root/Script/uploadervimeo.py", line 20, in <module>
    video_uri = v.upload (video)
  File "build/bdist.linux-i686/egg/vimeo/upload.py", line 28, in upload
  File "build/bdist.linux-i686/egg/vimeo/upload.py", line 58, in _perform_upload
  File "build/bdist.linux-i686/egg/vimeo/upload.py", line 110, in _make_pass
  File "build/bdist.linux-i686/egg/vimeo/client.py", line 75, in caller
  File "/usr/local/lib/python2.7/site-packages/requests/api.py", line 124, in put
    return request('put', url, data=data, **kwargs)
  File "/usr/local/lib/python2.7/site-packages/requests/api.py", line 57, in request
    return session.request(method=method, url=url, **kwargs)
  File "/usr/local/lib/python2.7/site-packages/requests/sessions.py", line 475, in request
    resp = self.send(prep, **send_kwargs)
  File "/usr/local/lib/python2.7/site-packages/requests/sessions.py", line 585, in send
    r = adapter.send(request, **kwargs)
  File "/usr/local/lib/python2.7/site-packages/requests/adapters.py", line 453, in send
    raise ConnectionError(err, request=request)
requests.exceptions.ConnectionError: ('Connection aborted.', error(104, 'Connection reset by peer'))

Fortunately however sometimes it works regularly but for me it is a big deal.
The script that I created to upload is as follows:

import vimeo

# VIMEO USER DATA
v = vimeo.VimeoClient(
    token="XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX",
    key="XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX",
    secret="XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX")

# Read local video name
read1 = open("/root/localvideoname.txt","r")
video = read1.read()
read1.close()

# Read online video name
read2 = open("/root/onlinevimeoname.txt","r")
onlinename = read2.read()
read2.close()

# Upload Process
video_uri = v.upload (video)

# Update online name
v.patch(video_uri, data={'name': onlinename})

# ID output Vimeo
print video_uri


Where am I wrong? your platform is Debian 6. Thanks in advance.

**Comment by Dashron** [comment]
We noticed some server issues that should be resolved. Are you still encountering this issue?

**Comment by paolocomis** [comment]
Thank you for your answer. Unfortunately, the problem occurs daily from 4 April 2016 (verified in my logs). I uploaded many videos a day and often all consecutively. Is there a way to fix it? I tried to add timeout and sleep even 5 minutes per video in scripts but the problem persists.

**Comment by Dashron** [comment]
In the response from the HTTP POST request to /me/videos, there is a field named "ticket_id". Can you log a handful of those ticket ids and post them here? We'll keep our team updated and try to track down the cause.

**Comment by Dashron** [comment]
cc @tommypenner

**Comment by paolocomis** [comment]
Hello, this morning I received an email from vimeo support (which in the meantime I had contacted). It seems that they had problems in the server.

The answer:

Hi there,
After my reply to you yesterday we pushed a fix for the upload servers That were timing out. Are you still having trouble with your requests failing / timing out today?
Sincerely,
Tommy P.

Yesterday in my logs I found no errors on 47 video upload. So I think this is not a problem of the python API but of Vimeo server.

Thanks anyway for your support and will keep you informed.
